### PR TITLE
Remove compilation warning fix data type format error

### PR DESCRIPTION
**Pull request by zhuyaliang**

### CODE REVIEW

**Comment by raveit65** [comment]
I found only a few warnings in previous builds but for most changes i do not see any warning.
Eg. where do you see `data type format error` ?

**Comment by zhuyaliang** [comment]
Compile warnings can appear using ```meson``` build
```
src/gpm-phone.c:394:18: warning: format '%i' expects argument of type 'int', but argument 4 has type 'guint' {aka 'unsigned int'} [-Wformat=]
```

**Comment by raveit65** [comment]
Hmm, normally i use auto-tools an i don't see the warnings with `--enable-compile-warnings=maximum`. So compiler flags are different in meson :/
Also with `meson setup build -Dprefix=/usr -Dwarning_level=everything` i don't see this warning.

**Comment by raveit65** [comment]
Ok, i see the warnings now.

File: applets/brightness/brightness-applet.c [hunk 1]
```diff
@@ -702,11 +705,6 @@ gpm_applet_popup_cb (GpmBrightnessApplet *applet, GdkEventButton *event)
 		return TRUE;
 	}
 
-	/* don't show the popup if brightness is unavailable */
-	if (applet->level == -1) {
-		return FALSE;
-	}
-
```
Comment (raveit65) [inline applets/brightness/brightness-applet.c:709]: You removing this condition because it doesn't work?

**Comment by raveit65** [comment]
I could confirm most of warnings with
```
export CFLAGS='-Wsign-compare -Wformat-signedness -Wformat=2 -Wswitch-default -Wcast-function-type -Wdeclaration-after-statement'

./autogen.sh --enable-compile-warnings=maximum
```
But not all. It's really difficult to review for me when cflags are different with auto-tools and meson. It took me several hours to understand why you want to change parts of the code.

File: applets/brightness/brightness-applet.c (hunk 1 above)
Comment (lukefromdc) [reply applets/brightness/brightness-applet.c:709]: It does in fact seem to fail: the popup slider comes up just fine with the applet on a desktop. Note however that this is can be useful in working on the applet, reducing the number of times the work has to be moved to a laptop. It's the entire applet not just the slider that has no function (outside being worked on) on a desktop, not just the slider.  Sound applet or status icon with no system sound support is a similar case though, and we use a null output there to allow the applet to render anyway and most importantly not crash. This helps users debug a lack of sound support, giving a shortcut to the menu which lists any options the system does support.

**Comment by lukefromdc** [comment]
This looks like at some point in the history of this code (presumably not when it was first written...) signed and unsigned integers got mixed up. Yes, this should be fixed.

**Review by lukefromdc — APPROVED** [review]
This is an all-in-one for a variety of code issues, builds and installs fine here. Looking at the code I see fixes for mixing signed and unsigned integers in debug messages, a prototype warning fix, and a style fix in `gpm_applet_create_popup` where we declare four variables together instead of two at the start and two later inline.

Looks like a good code cleanup to me, and doing this as an all-in-one means fewer PR's to deal with at the expensive of one more complex review, which is fine by me.

**Review by raveit65 — APPROVED** [review]
Code changes are looking good to me.
m-p-m runs fine on desktop machine.
Not tested on my notebook with full battery, brightness an power-management support as i do not run 1.27 on this device.

**Comment by raveit65** [comment]
@zhuyaliang 
Can you please post warnings cflags for next future PR if it so huge again? 
This will make reviewers life easier

**Comment by lukefromdc** [comment]
Anyone else want to review this or is it ready to go?

**Comment by raveit65** [comment]
I tested it again on my notebook and power management and brightness function are working.
Let's merge it.

**Pull request merged**